### ISSUE TITLE
2.5 and 2.6 beta / html - css and chtml: Horizontal bars use px instead of em 

### ISSUE THREAD

**Issue by gitcorn** [mathjax/MathJax#1290]
Horizontal bar thickness in fractions or as part of the root symbol are denoted in px instead of em.

This issue exists in html-css and common html. Tested on 2.5 and the new 2.6.1 beta. Firefox and Chrome.

Look at the three screenshots (all done with chtml on firefox):

first: regular mathjax.org site
second: mathjax.org but body font size changed to 44 px after math has been rendered
third: same as second but line thickness changed to 0.05 em scaling good for big and lil fonts.

![1](https://cloud.githubusercontent.com/assets/15322799/10727639/acaaa0e2-7bdb-11e5-9ea3-827edb11c0af.jpg)
![2](https://cloud.githubusercontent.com/assets/15322799/10727642/b56855ee-7bdb-11e5-9d06-5049cb4dff28.jpg)
![3](https://cloud.githubusercontent.com/assets/15322799/10727646/ba63e8e2-7bdb-11e5-8183-be845ab70e48.jpg)

Some might ask: "Why is this an issue?"

1. Dynamic resizing of fonts on website without re-rendering Math should be looking good (enough).
2. Mathjax-node chtml output should be size independent (within certain limits).
3. It might be necessary for future uses, like device independent pre-rendered html or even epub3. I use some js code within my browser to extract the chtml part (no more html-css) and embed it in a html file with a css file and woff fonts and get a reasonable sized directory for my purposes.

**Comment by dpvc** [comment]
Unfortunately, there are competing issues involved here.  You would like the output to be scale independent, which I fully understand.  On the other hand, browsers are very prone to drop thin horizontal and vertical lines, especially when the output is zoomed (see issues #1279 and #1095).  It turns out that when the line size is given in `px` rather than `em`, the browsers are much less likely to lose the lines, even at small zoom levels.  So we switched to px rather than em.  The loss of the lines in normal browser use (even at 100% zoom in some cases) outweighed the desire for scale independence for potential future use.

**Comment by gitcorn** [comment]
OK, I understand.

Before we close this one: 

Is the actual px to em ratio known for the new Mathjax font? I could use it for my purposes (and probably some MJ-node users too).

**Comment by dpvc** [comment]
> Is it possible to use a configuration switch for optional em usage?

I was going to suggest that as well.  That certainly could be done.

The CommonHTML output has not been fully integrated into MathJax-node yet, so the details of how it will get the pixel-to-em ratio is not set in stone.  The current prototype uses the ex-size to get the em-size, but that is not really sufficient, so an option to MathJax-node for em's may need to be added as well.

**Comment by gitcorn** [comment]
Alright, thx for information

I'm closing this issue.
